pynumdiff/optimize/_optimize.py

```diff
@@ -132,27 +140,28 @@ def _objective_function(point, func, x, dt, singleton_params, categorical_params
 
     :return: float, cost of this objective at the point
     """
+    # Short circuit if this hyperparam combo has already been queried, ~10% savings per #160
     key = sha1((''.join(f"{v:.3e}" for v in point) + # This hash is stable across processes. Takes bytes
                ''.join(str(v) for k,v in sorted(categorical_params.items()))).encode()).digest()
-    if key in cache: return cache[key] # short circuit if this hyperparam combo has already been queried, ~10% savings per #160
+    if key in cache: return cache[key]
 
+    # Query the differentiation method at this choice of hyperparameters
     point_params = {k:(v if search_space_types[k] == float else int(np.round(v)))
                         for k,v in zip(search_space_types, point)} # point -> dict
-    # add back in the singletons we're not searching over
+    # add back in singletons and categorical choices the Nelder-Mead isn't searching over
     try: x_hat, dxdt_hat = func(x, dt, **point_params, **singleton_params, **categorical_params) # estimate x and dxdt
     except np.linalg.LinAlgError: cache[key] = 1e10; return 1e10 # some methods can fail numerically
 
-    # evaluate estimate
-    if dxdt_truth is not None: # then minimize ||dxdt_hat - dxdt_truth||^2
-        if metric == 'rmse':
-            rms_rec_x, rms_x, rms_dxdt = evaluate.rmse(x, dt, x_hat, dxdt_hat, dxdt_truth=dxdt_truth, padding=padding)
+    # Evaluate estimate according to a loss function
+    if dxdt_truth is not None:
+        if metric == 'rmse': # minimize ||dxdt_hat - dxdt_truth||_2
+            rms_dxdt = evaluate.rmse(dxdt_truth, dxdt_hat, padding=padding)
             cache[key] = rms_dxdt; return rms_dxdt
         elif metric == 'error_correlation':
-            ec = evaluate.error_correlation(dxdt_hat, dxdt_truth, padding=padding)
+            ec = evaluate.error_correlation(dxdt_truth, dxdt_hat, padding=padding)
             cache[key] = ec; return ec
-    else: # then minimize [ || integral(dxdt_hat) - x||^2 + gamma*TV(dxdt_hat) ]
-        rms_rec_x, rms_x, rms_dxdt = evaluate.rmse(x, dt, x_hat, dxdt_hat, dxdt_truth=None, padding=padding)
-        cost = rms_rec_x + tvgamma*evaluate.total_variation(dxdt_hat, padding=padding)
+    else: # then minimize sqrt{2*Mean(Huber((x_hat- x)/sigma))}*sigma + gamma*TV(dxdt_hat)
+        cost = evaluate.robust_rme(x, x_hat, padding=padding) + tvgamma*evaluate.total_variation(dxdt_hat, padding=padding)
```

Comment:
This case is now a bit simplified in code here, although it's doing a fancier thing. The evaluation code improvements paying dividends.